core/pkg/sync-store/sync_store.go

```diff
@@ -0,0 +1,242 @@
+package sync_store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/open-feature/flagd/core/pkg/logger"
+	isync "github.com/open-feature/flagd/core/pkg/sync"
+	"github.com/open-feature/flagd/core/pkg/sync/file"
+	"github.com/open-feature/flagd/core/pkg/sync/kubernetes"
+	"go.uber.org/zap"
+)
+
+var (
+	regCrd  *regexp.Regexp
+	regFile *regexp.Regexp
```

Comment:
Do we support file syncs ? 🤔 

If so we need to rethink file read overhead. K8s syncs are backed by informer caches. This is not the case with files.

Reply:
we do not, I've included this as a feature for easy debugging and won't be in the final product

Reply:
Nice, thanks :)